sw/device/tests/crypto/aes_gcm_timing_test.c

```diff
@@ -0,0 +1,64 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/lib/crypto/drivers/aes.h"
+#include "sw/device/lib/crypto/impl/aes_gcm/aes_gcm.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+#include "sw/device/tests/crypto/aes_gcm_testutils.h"
+#include "sw/device/tests/crypto/aes_gcm_testvectors.h"
+
+/**
+ * Checks that decryption takes the same number of cycles regardless of tag.
+ *
+ * This test ensures that the AES-GCM tag check takes the exact same amount of
+ * time for all *incorrect* tags; if it didn't, for instance exiting early at
+ * the first incorrect byte, then an attacker could guess one byte at a time
+ * and observe the timing to see if the byte was correct.
+ *
+ * It's OK if decryption takes a different amount of time with a correct tag
+ * than with an incorrect one.
+ *
+ * @param test Test vector to use
+ * @param key_shares AES key shares
+ */
+static void test_decrypt_timing(aes_gcm_test_t test) {
+  // Call AES-GCM decrypt with an incorrect tag (first word wrong).
+  test.tag[0]++;
+  uint32_t cycles_invalid1 = call_aes_gcm_decrypt(test, false);
```

Comment:
Optional<Nit>: I might annotate with `/*tag_valid=*/false` for clarity, especially since `call_aes_gcm_decrypt` comes from another file.